configuration/resize-disk-image.md

```diff
@@ -1,113 +1,89 @@
 
-To grow the private disk image of an AppVM beyond this limit, [qvm-grow-private](/doc/dom0-tools/qvm-grow-private/) can be used:
+To grow the root or private disk image of an AppVM beyond this limit, `qvm-volume` can be used:
 
 ~~~
-qvm-grow-private <vm-name> <size>
+qvm-volume extend <vm_name>:root <size>
+~~~
+OR
+~~~
+qvm-volume extend <vm_name>:private <size>
 ~~~
 
-Note: Size is the target size (i.e. 4096MB or 16GB, ...), not the size to add to the existing disk. 
-
-### Shrinking private disk image (Linux VM, R4.0)
-
-1.  Create a new qube with smaller disk using Qube Manager or `qvm-create`
-2.  Move data to the new qube using `qvm-copy` or OS utilities
-3.  Delete old qube using Qube Manager or `qvm-remove`
-
-### Shrinking private disk image (Linux VM, R3.2)
-
-**This operation is dangerous and this is why it isn't available in standard Qubes tools.
-If you have enough disk space, it is safer to create a new VM with a smaller disk and move the data.**
+Note: Size is the target size (i.e. 4096MB or 16GB, ...), not the size to add to the existing disk.
 
-The basic idea is to:
+### Expand disk image (R3.2)
 
-1.  Shrink filesystem on the private disk image.
-2.  Then shrink the image.
+1048576 MB is the maximum size which can be assigned to storage through Qubes Manager.
 
-Ext4 does not support online shrinking, so it can't be done as conveniently as growing the image.
-Note that we don't want to touch the VM filesystem directly in dom0 for security reasons. 
-First you need to start VM without `/rw` mounted. 
-One possibility is to interrupt its normal startup by adding the `rd.break` kernel option:
+To grow the private disk image of an AppVM beyond this limit, `qvm-grow-root` or [qvm-grow-private](/doc/dom0-tools/qvm-grow-private/) can be used:
 
 ~~~
-qvm-prefs -s <vm-name> kernelopts rd.break
-qvm-start --no-guid <vm-name>
+qvm-grow-root <vm-name> <size>
 ~~~
-
-And wait for qrexec connect timeout (or simply press Ctrl-C). 
-Then you can connect to VM console and shrink the filesystem:
-
+OR
 ~~~
-sudo xl console <vm-name>
-# you should get dracut emergency shell here
-mount --bind /dev /sysroot/dev
-chroot /sysroot
-mount /proc
-e2fsck -f /dev/xvdb
-resize2fs /dev/xvdb <new-desired-size>
-umount /proc
-exit
-umount /sysroot/dev
-poweroff
+qvm-grow-private <vm-name> <size>
 ~~~
 
-Now you can resize the image:
-
-~~~
-truncate -s <new-desired-size> /var/lib/qubes/appvms/<vm-name>/private.img
-~~~
+Note: Size is the target size (i.e. 4096MB or 16GB, ...), not the size to add to the existing disk. 
 
-**It is critical to use the same (or bigger for some safety margin) size in truncate call compared to resize2fs call.
-Otherwise you will lose your data!** 
-Then reset kernel options back to default:
+### Resize a StandaloneVM Root Image
 
-~~~
-qvm-prefs -s <vm-name> kernelopts default
-~~~
+Another way to increase the size of is to turn your TemplateVM into a StandaloneVM.
```

Comment:
Well, creating StandaloneVM does not make it larger... But it may be more convenient if you need a lot of extra software in just one VM. I think this sentence needs to be reworded.